### ISSUE TITLE
Allow updates similar to Rails enums

### ISSUE THREAD

**Issue by mthurman** [pboling/flag_shih_tzu#58]
I swapped out some enums I was using for some flags. Especially in tests where I had to set up specific models, I was changing code like this: `model.update(enum_field: :enum_value)` to `model.update(flags: [:flag1, :flag2])` which causes a `null` value to get sent to the db.

Instead, I ended up doing something like `model.update(flags: 3)` with a magic number.

It looks like it might be possible to do something similar to enums (https://github.com/rails/rails/blob/a9dc45459abcd9437085f4dd0aa3c9d0e64e062f/activerecord/lib/active_record/enum.rb#L165). Is there any interest in that?

Or any other recommendations to avoid magic numbers when setting multiple flags at once (without having to have one line of code per flag)?

**Comment by pboling** [comment]
Yes, that would be beautiful!  Please add some tests if you end up working on this.  The master build on travis is now passing!

**Comment by AlexDWu** [comment]
relevant? https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/attribute_methods/serialization.rb

**Comment by pboling** [comment]
Fixed in 6175c5ab116f3f35cb7b9a01993a87bdbffe1381

- `record.update!(flags: [:warpdrive, :shields])` now works.
- Array assignment means “set this flag column to exactly these selected flags.”
- Hash assignment means “update only these named flags and preserve the rest.”
- Raw integer assignment still works.
- Custom flag columns get the same behavior through generated writers like `features=` / `bits=`.
- Replaced `String#delete_prefix` in the `not_` flag path with Ruby 2.4-compatible slicing.
